certbot-apache/certbot_apache/interfaces.py

```diff
@@ -186,6 +212,26 @@ class DirectiveNode(ParserNode):
     parameters: Tuple[str, ...]
 
     """
+    # pylint: disable=too-many-arguments, super-init-not-called
+    def __init__(self, name, parameters=(), ancestor=None, filepath="",
+                 dirty=False, enabled=True):  # pragma: no cover
```

Comment:
There's probably a balance to be struck here, but I'm personally leaning towards removing a lot of/all of these default values. I think default values makes it easy for you to forget to add a parameter which is especially true where these functions take so many parameters.

What do you think?

Reply:
I agree. The only ones that I think might make sense to keep optional would be `dirty` and `enabled`.

Reply:
... then again, making all the arguments required, makes the implementation of https://github.com/certbot/certbot/pull/7330#issuecomment-522795095 much easier, as all the expected parameters are also required.

Reply:
Right. I don't care too much about this and default parameters are fairly common in Python code, but I think:

1. There are times when we'll want to remember to set directives even those like `dirty` and `enabled` to a different value (such as adding a new directive to the config and parsing a disabled block respectively) and writing the code such that we are forced to set them could help prevent bugs.
2. Creation of these objects will happen almost entirely inside the implementations themselves. The pain of doing this won't really escape outside of the class I think.